onlinejudge/type.py

```diff
@@ -343,15 +335,12 @@ def get_url(self) -> str:
         raise NotImplementedError
 
     @abstractmethod
-    def get_problem(self) -> Problem:
+    def download_problem(self) -> Problem:
```

Comment:
```suggestion
    def download_problem(self, *, session: Optional[requests.Session] = None) -> Problem:
```

`get_*` と違って `download_*` はネットワークアクセスを含むことを示唆するような名前であるので、引数でそのためのセッション情報を渡す必要があります。